File: Data Structure&Algorithm/meeting/meituan/demo4.py
def main():
    n,a,b=list(map(int,input().split()))
    arr=[]
    ans1,ans2=0,0
    for i in range(n):
        k1,k2=list(map(int,input().split()))
        arr.append((k1,k2))
    arr.sort(key=lambda x:x[0],reverse=True)
    for i in range(a):
        ans1+=arr[i][0]
    brr=arr[a:]
    brr.sort(key=lambda x:x[1],reverse=True)
    for i in range(b):
        ans1+=brr[i][1]
    #2
    arr.sort(key=lambda x:x[1],reverse=True)
    for i in range(b):
        ans2+=arr[i][0]
    brr=arr[a:]
    brr.sort(key=lambda x:x[0],reverse=True)
    for i in range(a):
        ans2+=brr[i][0]
    ans=max(ans1,ans2)
    print(ans)
main()

File: Data Structure&Algorithm/meeting/tengxun/before/demo4.cpp
#include <iostream>
#include <algorithm>
using namespace std;
int arr[100002];
bool cmp(int i, int j) { return (i > j); } //降序排列
int main()
{
    int n;
    cin >> n;
    int i;
    for (i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    if(n%2==1){
        arr[i]=0;
        n++;
    }
    sort(arr, arr + n, cmp);
    int ans=0;
    for (int i = 0; i < n-1; i+=2)
    {
        ans+=arr[i]-arr[i+1];
    }
    cout<<ans<<endl;
    return 0;
}